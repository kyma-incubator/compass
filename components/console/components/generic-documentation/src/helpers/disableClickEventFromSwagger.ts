export function disableClickEventFromSwagger() {
  const beginOfSwaggerClickFunctionAsString = `for (var t = e.target; t && "A" !== t.tagName;)`;

  const originalEventListener = window.addEventListener;
  window.addEventListener = (
    type: string,
    fn: (...args: any[]) => any,
    options: any,
  ) => {
    if (type === 'click') {
      const fnAsString = fn.toString();
      if (fnAsString.includes(beginOfSwaggerClickFunctionAsString)) {
        return;
      }
    }
    return originalEventListener(type, fn, options);
  };
}
