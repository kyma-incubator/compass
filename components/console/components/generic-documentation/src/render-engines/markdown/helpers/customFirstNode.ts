import { Source } from '@kyma-project/documentation-component';
import { plugins } from '@kyma-project/dc-markdown-render-engine';
import { headingPrefix } from './headingPrefix';

type Header = plugins.Header;

export function customFirstNode(
  source: Source,
  toKebabCase: (str: string) => string,
): Header | undefined {
  const title: string =
    source.data && source.data.frontmatter && source.data.frontmatter.title;

  if (!title) {
    return;
  }

  return {
    title,
    id: toKebabCase(headingPrefix(source)),
    level: 'doc-title',
    source,
  };
}
