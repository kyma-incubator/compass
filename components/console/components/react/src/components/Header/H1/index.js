import styled from 'styled-components';
import { rem } from 'polished';
import { media } from '../../../commons';

const H1 = styled.h1`
  font-size: 2em;
  line-height: 1.2;

  ${media.giant`
    font-size: 3em;
    line-height: 1.025;
  `} ${media.desktop`
    font-size: 2.5em;
    line-height: 1.1;
  `};
`;

export default H1;
