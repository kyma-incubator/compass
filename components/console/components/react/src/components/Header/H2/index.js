import styled from 'styled-components';
import { media } from '../../../commons';

const H2 = styled.h2`
  font-size: 1.625em;
  line-height: 1.15384615;

  ${media.giant`
    font-size: 2.25em;
    line-height: 1.25;
  `} ${media.desktop`
    font-size: 2em;
    line-height: 1.25;
  `};
`;

export default H2;
