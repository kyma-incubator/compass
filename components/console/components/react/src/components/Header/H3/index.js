import styled from 'styled-components';
import { media } from '../../../commons';

const H3 = styled.h3`
  font-size: 1.375em;
  line-height: 1.13636364;

  ${media.giant`
    font-size: 1.75em;
    line-height: 1.25;
  `} ${media.desktop`
    font-size: 1.5em;
    line-height: 1.25;
  `};
`;

export default H3;
