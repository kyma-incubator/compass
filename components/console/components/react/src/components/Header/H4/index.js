import styled from 'styled-components';
import { media } from '../../../commons';

const H4 = styled.h4`
  font-size: 1.125em;
  line-height: 1.11111111;

  ${media.desktop`
    line-height: 1.22222222;
  `};
`;

export default H4;
