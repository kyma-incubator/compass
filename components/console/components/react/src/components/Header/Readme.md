```jsx
  <Header>
    <H1>Header h1</H1>
    <H2>Header h2</H2>
    <H3>Header h3</H3>
    <H4>Header h4</H4>
  </Header>
```
