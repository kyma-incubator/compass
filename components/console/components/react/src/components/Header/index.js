import styled from 'styled-components';
import { media } from '../../commons';

const Header = styled.header`
  font-size: 100%;
  font-size-adjust: 0.5;
  font-weight: normal;
  color: #32363a;
  margin: ${props => (props.margin ? props.margin : '0')};
`;

export default Header;
