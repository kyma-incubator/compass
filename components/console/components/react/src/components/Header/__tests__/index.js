import React from 'react';
import Header from '../index';
import 'jest-styled-components';

import { renderWithTheme } from 'test_setup/helpers';

describe('Header Element', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(<Header>A Header tag</Header>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
