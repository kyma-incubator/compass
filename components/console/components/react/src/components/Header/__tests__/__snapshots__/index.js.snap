// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Element renders correctly 1`] = `
.c0 {
  font-size: 100%;
  font-size-adjust: 0.5;
  font-weight: normal;
  color: #32363a;
  margin: 0;
}

<header
  className="c0"
>
  A Header tag
</header>
`;
